### PR TITLE
Ensure instances of auto types are initialized

### PR DESCRIPTION
**Pull request by sjaeckel**
Follow-up of #1913

### CODE REVIEW

File: src/ui/window.c [hunk 1]
```diff
@@ -1389,14 +1389,8 @@ win_print_incoming(ProfWin* window, const char* const display_name_from, ProfMes
...
     {
-        auto_char char* enc_char;
         ProfChatWin* chatwin = (ProfChatWin*)window;
-
-        if (chatwin->incoming_char) {
-            enc_char = strdup(chatwin->incoming_char);
-        } else {
-            enc_char = get_show_char(message->enc);
-        }
+        auto_char char* enc_char = chatwin->incoming_char ? strdup(chatwin->incoming_char) : get_show_char(message->enc);
```
Comment (jubalh) [inline src/ui/window.c:1393]: why no get_enc_char() here?

File: src/xmpp/message.c [hunk 2]
```diff
@@ -98,6 +98,20 @@ static xmpp_stanza_t* _ox_openpgp_signcrypt(xmpp_ctx_t* ctx, const char* const t
 
 static GHashTable* pubsub_event_handlers;
 
+gchar*
+get_display_name(const ProfMessage* const message, gboolean* flags)
```
Comment (jubalh) [inline src/xmpp/message.c:102]: I'm unsure why flags is a `gboolean` here. In the calling functions it's an `int` and:
```
// flags : 1st bit =  0/1 - me/not me. define: NO_                                                              
//         2nd bit =  0/1 - date/no date. define: NO_DATE
//         3rd bit =  0/1 - eol/no eol. define: NO_EOL
//         4th bit =  0/1 - color from/no color from. define: NO_COLOUR_FROM
//         5th bit =  0/1 - color date/no date. define: NO_COLOUR_DATE
//         6th bit =  0/1 - trusted/untrusted. define: UNTRUSTED
```

File: src/ui/window.c (hunk 1 above)
Comment (sjaeckel) [reply src/ui/window.c:1393]: Because I missed it :)

File: src/xmpp/message.c (hunk 2 above)
Comment (sjaeckel) [reply src/xmpp/message.c:102]: Funny how the crippled glib types made this not even spit a warning ...

**Review by jubalh — APPROVED** [review]

**Pull request merged**